api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -36,12 +54,110 @@
+        this.cohortDao = cohortDao;
+        this.participantCohortStatusDao = participantCohortStatusDao;
+        this.bigQueryService = bigQueryService;
+        this.participantCounter = participantCounter;
+    }
+
+    /**
+     * Create a cohort review per the specified workspaceId, cohortId, cdrVersionId and size. If participant cohort status
+     * data exists for a review or no cohort review exists for cohortReviewId then throw a
+     * {@link BadRequestException}.
+     *
+     * @param workspaceId
+     * @param cohortId
+     * @param cdrVersionId
+     * @param size
+     * @return
+     */
+    @Override
+    public ResponseEntity<org.pmiops.workbench.model.CohortReview> createCohortReview(
+            Long workspaceId, Long cohortId, Long cdrVersionId, Integer size) {
+        if (size <= 0 || size > MAX_REVIEW_SIZE) {
+            throw new BadRequestException("Invalid Request: Cohort Review size must be between 0 and " + MAX_REVIEW_SIZE);
+        }
+        CohortReview cohortReview = cohortReviewDao.findCohortReviewByCohortIdAndCdrVersionId(cohortId, cdrVersionId);
+        if (cohortReview == null) {
+            throw new BadRequestException("Invalid Request: Cohort Review does not exist for cohortId: "
+                    + cohortId + ", cdrVersionId: " + cdrVersionId);
+        }
+        if(cohortReview.getReviewSize() > 0) {
+            throw new BadRequestException("Invalid Request: Cohort Review already created for cohortId: "
+                    + cohortId + ", cdrVersionId: " + cdrVersionId);
+        }
+
+        CohortDefinition definition = cohortDao.findCohortByCohortIdAndWorkspaceId(cohortId, workspaceId);
+        if (definition == null) {
+            throw new BadRequestException("Invalid Request: No Cohort definition matching cohortId: "
+                    + cohortId + ", workspaceId: " + workspaceId);
+        }
+        SearchRequest request = new Gson().fromJson(definition.getCriteria(), SearchRequest.class);
+        QueryResult result = bigQueryService.executeQuery(bigQueryService.filterBigQueryConfig(
+                participantCounter.buildParticipantIdQuery(request, size)));
+        Map<String, Integer> rm = bigQueryService.getResultMapper(result);
+
+        List<ParticipantCohortStatus> participantCohortStatuses = new ArrayList<>();
+        for (List<FieldValue> row : result.iterateAll()) {
+            participantCohortStatuses.add(
+                    new ParticipantCohortStatus()
+                            .participantKey(
+                                    new ParticipantCohortStatusKey(
+                                            cohortReview.getCohortReviewId(),
+                                            bigQueryService.getLong(row, rm.get("person_id"))))
+                            .status(CohortStatus.NOT_REVIEWED));
+        }
+
+        cohortReview.setReviewSize(size);
+        cohortReview.setParticipantCohortStatuses(participantCohortStatuses
```

Comment:
Do you need to set the review status on the review? Set reviewed count to zero?

Reply:
Done.